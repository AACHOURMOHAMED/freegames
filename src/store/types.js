const GET_GAMES = 'get_games';

export default GET_GAMES;
