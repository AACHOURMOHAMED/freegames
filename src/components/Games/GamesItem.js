import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import classes from './gameItem.module.scss';

const GamesItem = (props) => {
  const {
    id, title, released, image, gener,
  } = props;
  return (
    <NavLink to={`/details/${id}`} className={classes.card}>
      <img src={image} alt={title} />
      <div className={classes.back} />
      <div className={classes.card_info}>
        <ul key={id}>
          <li>{title}</li>
          <li>{gener}</li>
          <li>{released}</li>
        </ul>
      </div>
    </NavLink>
  );
};

export default GamesItem;

GamesItem.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  released: PropTypes.string.isRequired,
  gener: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};
