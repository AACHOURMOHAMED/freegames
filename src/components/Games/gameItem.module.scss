.card {
    position: relative;
    display: grid;
    width: 100%;
    height: 100px;
    img{
        position: absolute;
        width: 100%;
        height: 100%;
        z-index: -11;
        object-fit: cover;
    }
    
    .back {
        position: absolute;
        width: 100%;
        height: 100%;
        background-color: rgba(0, 0, 0, 0.616);
        z-index: -2;
    }
    .card_info {
        display: flex;
        align-items: center;
        ul {
            list-style: none;
            padding-left: 20px;
            li {
                color: #c1c1c1;
                font-weight: normal;
                font-size: 11px;
                &:first-child {
                    color: #fff;
                    font-weight: bold;
                    font-size: 14px;
                }
            }
        }
    }
    &:first-child {
        grid-column: 1 / span 2;
        height: 250px;
        .card_info{
            li {
                color: #c1c1c1;
                font-weight: normal;
                font-size: 14px;
                &:first-child {
                    color: #fff;
                    font-weight: bold;
                    font-size: 25px;
                }
            }
        }
    }
    &:last-child {
        grid-column: 1 / span 2;
    }
}

// add desktop versoin
@media (min-width: 768px) {
    .card {
        height: 250px;
        &:first-child {
            height: 450px;
        }

        .card_info {
            ul {
                li {
                    font-size: 18px;
                    &:first-child {
                        font-size: 30px;
                    }
                }
            }
        }
    }
}