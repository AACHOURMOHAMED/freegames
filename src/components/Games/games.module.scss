.search {
  position: absolute;
  display: flex;
  justify-content: space-between;
  width: 100%;
  padding: 10px;
  z-index: 999;
  height: 50px;

  input {
    width: 100%;
    padding: 2px 8px;
    border-radius: 5px;
    border: none;

    &:focus {
      outline: none;
    }
  }

  &:focus {
    outline: none;
  }
}

.card {
  display: grid;
  grid-template-columns: 1fr 1fr;
  position: relative;

  .notFound {
    color: #fff;
    margin-top: 80px;
    margin-left: 40px;

    h1 {
      font-size: 30px;
      font-weight: bold;
    }
  }
}
