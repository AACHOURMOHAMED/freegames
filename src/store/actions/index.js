/* eslint-disable import/prefer-default-export */
import { createAsyncThunk } from '@reduxjs/toolkit';
import axios from 'axios';
import * as types from '../types';

const URL = 'https://api.rawg.io/api';

export const getGames = createAsyncThunk(types.GET_GAMES,
  async () => {
    const { data } = await axios.get(`${URL}/games?key=b83701e00bdf4e8fa7e7e7c8f6c1d9cb`);
    const games = data.results.map((game) => ({
      id: game.id,
      title: game.name,
      image: game.background_image,
      gener: game.genres.map((genre) => genre.name).join(', '),
      released: game.released,
      platphorm: game.platforms.map((platform) => platform.platform.name).join(', '),
      year_start: game.released.split('-')[0],
      playtime: game.playtime,
      updated: game.updated,
      suggestions_count: game.suggestions_count,
      rating: game.rating,
      tags: game.tags.map((tag) => tag.name).join(', ').split(', ', 5).join(', '),
      stores: game.stores.map((store) => store.store.name).join(', '),
    }));
    return games;
  });
