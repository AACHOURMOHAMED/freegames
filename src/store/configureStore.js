import { configureStore, applyMiddleware, compose } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import rootReducer from './reducers';

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = configureStore({
  reducer: rootReducer,
  composeEnhancers: composeEnhancers(applyMiddleware(thunk, logger)),
});

export default store;
