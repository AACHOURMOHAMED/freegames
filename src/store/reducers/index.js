import { combineReducers } from '@reduxjs/toolkit';
import games from './games_reducer';

const rootReducer = combineReducers({
  games,
});

export default rootReducer;
