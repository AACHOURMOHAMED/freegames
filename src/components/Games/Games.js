/* eslint-disable no-nested-ternary */
import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getGames } from '../../store/actions';
import GamesItem from './GamesItem';
import classes from './games.module.scss';

const Games = () => {
  const [search, setSearch] = useState('');
  const [loading, setLoading] = useState(true);
  const games = useSelector((state) => state.games);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getGames());
    setLoading(false);
  }, []);

  const handleChange = (e) => {
    setSearch(e.target.value);
  };
  const filteredGames = games.filter((game) => game.title.toLowerCase()
    .includes(search.toLowerCase()));

  return (
    <div className={classes.card}>
      <div className={classes.search}>
        <input type="text" placeholder="Search..." onChange={handleChange} value={search} />
      </div>
      {
        !loading ? (
          filteredGames.length > 0 ? (
            filteredGames.map((game) => (
              <GamesItem
                key={game.id}
                id={game.id}
                title={game.title}
                released={game.released}
                image={game.image}
                gener={game.gener}
              />
            ))
          ) : (
            <div className={classes.notFound}>
              <h1>No games found</h1>
            </div>
          )

        ) : (
          <div className="spinner-border text-primary" role="status">
            <span className="sr-only">Loading...</span>
          </div>

        )
        }
    </div>

  );
};

export default Games;
