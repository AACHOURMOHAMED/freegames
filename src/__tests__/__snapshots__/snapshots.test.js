import TestRenderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../store/configureStore';
// import all the components you want to test
import Games from '../../components/Games/Games';
import GameDetail from '../../components/Details/Gamedetail';

describe('test screenshots for all components', () => {
//   const getState = () => ({ price: 0.145947, ticket: 'EUR/USD' });

  test('matches Games snapshot', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <Router>
          <Games />
        </Router>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches GameDetails snapshot', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <Router>
          <GameDetail />
        </Router>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
