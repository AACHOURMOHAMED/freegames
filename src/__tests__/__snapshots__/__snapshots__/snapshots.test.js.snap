// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test screenshots for all components matches GameDetails snapshot 1`] = `
<div
  className="spinner-border text-primary"
  role="status"
>
  <span
    className="sr-only"
  >
    Loading...
  </span>
</div>
`;

exports[`test screenshots for all components matches Games snapshot 1`] = `
<div
  className="card"
>
  <div
    className="search"
  >
    <input
      onChange={[Function]}
      placeholder="Search..."
      type="text"
      value=""
    />
  </div>
  <div
    className="spinner-border text-primary"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;

exports[`test screenshots for all components matches Header snapshot 1`] = `
<header>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
  >
    <div
      className="back"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 11H6.414l5.293-5.293-1.414-1.414L2.586 12l7.707 7.707 1.414-1.414L6.414 13H21z"
        />
      </svg>
    </div>
  </a>
</header>
`;

exports[`test screenshots for all components matches NotFound snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <h1>
        Not Found
      </h1>
    </div>
  </div>
</div>
`;
